### ISSUE TITLE
linkchecker exclude only with regex do not include with regex

### ISSUE THREAD

**Issue by estatistics** [linkchecker/linkchecker#606]
## Summary

linkchecker has only the possibility to exclude certain links with regex but not include certain links with regex. 

eg. linkchecker --include *viewlinksonlywithregex

linkchecker has only the possibility to exclude certain links with regex but not include certain links with regex. 

## Steps to reproduce

 1. linkchecker has only the possibility to exclude certain links with regex but not include certain links with regex. 
2.  linkchecker --include *viewlinksonlywithregex
 3. 

## Actual result
 linkchecker --exclude *viewlinksonlywithregex
## Expected result
 linkchecker --include *viewlinksonlywithregex
## Environment

 * Operating system: 
 * Linkchecker version: 
 * Python version: 
 * Install method:

**Comment by Mennaruuk** [comment]
Yeah, I've been having the same feature request. It'd be amazing if there's an option to only scan URLs matching a regex. I tried doing the inverse, that is, ignore URLs that [do not match a regex](https://stackoverflow.com/a/164419), thus only yielding the URLs that match that regex, but when I did that, I got 0 results.

`linkchecker --verbose --no-warnings --file-output=csv --ignore-url=r"(?!https://www.nytimes.com/2015[^/]+(?=/$|$))"`

```
Statistics:
Downloaded: 0B.
No statistics available since no URLs were checked.

That's it. 0 links in 0 URLs checked. 0 warnings found. 0 errors found.
Stopped checking at 2022-03-12 05:32:03-004 (0.00 seconds)
```

**Comment by cjmayo** [comment]
This is tied into the concept of internal and external links.

If not setting up to check external links (the default), and OK to check all internal links, this can be achieved by adding URL patterns to internal links with `internlinks` in the linkcheckerrc configuration file:
https://linkchecker.github.io/linkchecker/man/linkcheckerrc.html#filtering

If checking external links (`--check-extern` or `checkextern=1`), the above idea of using `--ignore-url` with a negative match does work - but raw (r strings) can't be passed as far as I can see - the regex must start with a `!`, which for me on Linux (Bash) needs escaping with a `\` i.e. `--ignore-url=\!...`. `-Dcmdline` can be used to check the value of `externlinks` and observe a strict pattern with `'negate': True,`.

**Comment by cjmayo** [comment]
If the pattern is "including" internal links then `--ignore-url` can be used.
